### PR TITLE
fix: stark sig auth account interaction

### PR DESCRIPTION
**Pull request by Orland0x**
To simply user actions, we removed support for alternate account implementations and now only support the standard introduced by SNIP-6.

this removes issue #554

### CODE REVIEW

File: starknet/src/tests/mocks/account.cairo [hunk 1]
```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
```
Comment (Orland0x) [inline starknet/src/tests/mocks/account.cairo]: im not sure whether there is a way to add a contract in a dependency to the compilation artifacts. So just directly added the account  code here.

Comment (Orland0x) [reply starknet/src/tests/mocks/account.cairo]: figured out how to do it

File: starknet/src/interfaces/i_account.cairo [hunk 2]
```diff
@@ -15,18 +15,3 @@ trait AccountABI<TState> {
     fn is_valid_signature(self: @TState, hash: felt252, signature: Array<felt252>) -> felt252;
```
Comment (pscott) [inline starknet/src/interfaces/i_account.cairo]: could we use the OZ library for the account interface? Or maybe there's a SNIP library somewhere we can use?

**Review by pscott — APPROVED** [review]
utACK

feelsgoodman.jpg

File: starknet/src/interfaces/i_account.cairo (hunk 2 above)
Comment (Orland0x) [reply starknet/src/interfaces/i_account.cairo]: yep imported from oz

**Pull request merged**